### ISSUE TITLE
No match for FieldRef.Name(split) in id: string

### ISSUE THREAD

**Issue by nanxue2023** [facebookresearch/large_concept_model#15]
Hello together, I get the following error when execute:
```
No match for FieldRef.Name(split) in id: string 
url: string 
title: string
text: string
text_sentences: list<element: string>
text_sentences_sonar_emb: list<element: fixed_size_list<element: float>[1024]>
```
I use pdb to print the stack information and find that the program run to line 1087 of parquet_utils.py `fragments = list(dataset._dataset.get_fragments(filter=dataset._filter_expression))`
I can't fix this. Could u help me to figure out?

**Comment by hiskuDN** [comment]
@elbayadm gave a good answer here, maybe it'll help. https://github.com/facebookresearch/large_concept_model/issues/9#issuecomment-2573568281

**Comment by nanxue2023** [comment]
@hiskuDN Thx!!!! https://github.com/facebookresearch/large_concept_model/issues/9#issuecomment-2569754348 helps me solve the problem but I meet a new error: **'pyarrow.lib.ListScalar' object has no attribute 'to'**. The error appears in line 191 `embs = [x.to(self.gang.device).to(dtype) for x in batch[col_name]]` in dataloader.py
I replace `embs = [x.to(self.gang.device).to(dtype) for x in batch[col_name]]`  with the following:
```
batch_py = torch.tensor(batch[col_name].to_pylist())
embs = [x.to(self.gang.device).to(dtype) for x in batch_py]
```
Another error occurs! **error=expected sequence of length 25 at dim 1 (got 26)**
😭

**Comment by hiskuDN** [comment]
What dataset are you using? Are you pytorch for data handling at any point in the pipeline?

**Comment by nanxue2023** [comment]
wikipedia dataset. I use this code to process the data:
```
import pandas as pd

dataset_path = "/content/large_concept_model/sample_data/0_a25e918a7789ecfa_0_0.parquet" # dataset path 

df = pd.read_parquet(dataset_path) # load dataset in pandas df

df['split'] = 'train' # adding 'split' column to dataset because of the missing split column

df.to_parquet(dataset_path) # convert dataset to parquest again
```

**Comment by hasanyazarr** [comment]
@nanxue2023 This was my solution to solve that error but I think it is outdated and seems not correct. So, you can check https://github.com/facebookresearch/large_concept_model/issues/23#issuecomment-2609392617 comment. It solves these errors and starts training model.
